planet-kaggle-blog.md

```diff
@@ -0,0 +1,69 @@
+
+Secondly, a small portion of the TIFF and JPEG chips were blatantly disimilar.
+
+<p align="center"><img src="imgs/broken.jpg" alt="Bad chips" height=550px /></p>
+<p align="center">source: <a href="https://www.kaggle.com/c/planet-understanding-the-amazon-from-space/discussion/32453">Heng CherKeng</a></p>
+
+The left hand column displays the JPEG chip, while the right hand columns display the TIFF chip under two different visualizations. In each case, the tags indicate the chip has, among others, the `road` tag. We can clearly see that in the TIFF chips were taken at some other time or location and clearly do not have a road within the image. If we teach a model that a image which does not have a road should be labeled road, this is a problem.
+
+In almost all these cases, the chip that had the incorrect label was the TIFF chip. Our suspicion is that the crowd sourced labelers were by and large generating tags using only the JPEG chips. While the labelers were likely given both file versions for the chips, viewing the JPEG version of the chips is easier. The TIFF images required some normalization to make it visible to the human eye. As an unfortunate result, any discrepancies between the file versions favored the JPEG chips. Although the exact extent of misconfigured labels between the file types is unclear, dozens of such conflicts were easily spotted throughout the dataset. The resulting noise proved to be a major challenge in producing accurate results with even our most complex models using either TIFF files.
+
+### Discussion
+
+Many techniques that we implemented in Raster Vision and trialed did not increase our F2 score on the Amazon dataset. In several cases, we believe that the F2 score metric had unintentional effects on what constituted an accurate result. For example, we tried creating separate networks with different output layers for atmospheric tags versus the other tags. As there can be only one correct atmospheric tag, it seemed that a sigmoid layer would output better predictions than a softmax layer. Although counterintuitive to our expectation, this did not yield improved results. The metric places greater value on predicions with both correct and incorrect labels more than predictions which do not have the correct label at all. We also had experiments that artificially boosted the presence of rarer labels during training. We suspect that the F2 score simply did not place as much value on the performance for infrequent tags. Even if our predictions were more accurate for rare tags, it came at the cost of our overall F2 score across all the tags.
```

Comment:
'Many techniques that we implemented in Raster Vision and trialed' -> 'Many techniques we tried'

Reply:
'As there can be only one correct atmospheric tag, it seemed that a sigmoid layer would output better predictions than a softmax layer.' It's the other way around -- a softmax layer picks a single tag out of a set.

Reply:
'We also had experiments that artificially boosted the presence of rarer labels during training.' I don't think this helped, even when looking at the individual tag scores.